tools/mpremote/mpremote/transport.py

```diff
@@ -63,6 +63,9 @@ def _convert_filesystem_error(e, info):
         return FileExistsError(info)
     if "OSError" in e.error_output and "ENODEV" in e.error_output:
         return FileNotFoundError(info)
+    if "OSError" in e.error_output and "EINVAL" in e.error_output:
+        return ValueError(info)
```

Comment:
These are not just arbitrary; the rest of these cases are the established python errno mapping for that errno value. See https://docs.python.org/3/library/errno.html.

Reply:
EINVAL should map to `OSError(errno.EINVAL, info)`, not `ValueError(info)`.

Reply:
All of the other exceptions it currently returns are ones that would be caught by `except OSError:`; returning an exception that is no longer a "corresponding OSError-derived exception" as documented breaks the function's type contract.

Reply:
Makes indeed sense to keep it to a subclass of OSError